src/mongocrypt-kms-ctx.c

```diff
@@ -50,6 +50,7 @@ typedef struct {
 #define SHA256_LEN 32
 #define DEFAULT_HTTPS_PORT "443"
 #define DEFAULT_KMIP_PORT "5696"
+static const int kms_max_attempts = 10;
```

Comment:
```suggestion
static const int kms_max_attempts = 3;
```

Suggest using a max of 3. I noted rationale in scope (avoid a long worst-case, agree with AWS defaults)